diff --git a/activity/activity.info b/activity/activity.info
index b44f2aa..dc17061 100644
--- a/activity/activity.info
+++ b/activity/activity.info
@@ -2,7 +2,15 @@
 name = NimGame
 activity_version = 3
 bundle_id = org.sugarlabs.NimGame
+release_date = 2024-02-16
 exec = sugar-activity3 Nimactivity.Nimgame
 icon = activity-nimgame
 license = GPLv3+
+metadata_license = CC0-1.0
 repository = https://github.com/SudoSu-bham/NimGame
+summary = A mathematical Sugar application
+description = The Game of Nim is a two-player mathematical game of strategy. Players take turns removing objects from the heaps according to specific rules. The objective of the game is to avoid being the player to take the last object. 
+url = https://github.com/SudoSu-bham/NimGame.git
+tags = Education
+update_contact = https://github.com/SudoSu-bham
+developer_name = SudoSu-bham
\ No newline at end of file

